### PR TITLE
feat: Adding trace option 'on-all-retries'

### LINKED ISSUE #21860: [Feature] add a trace option in the playwright conf to trace on all retries
Hi!
Today we use the option 'retain-on-failure' to get traces on all of our retries (actually all of the failing runs) but this also mean that we are tracing the whole time and this consumes resources. We run our tests in a pipeline with limited resources. In these cases it would be very handy with an option that only enables retries on retries but also traces in all the retries. 'on-first-retry' and 'retry-with-trace' only do tracing in the first retry. It would be ideal if 'retry-with-trace' did trace all the retries. Also since it's not mentioned here:
https://playwright.dev/docs/test-configuration#record-test-trace
I feel I can't file a bug for it.

I hope to see this anytime soon.

Cheers!
// Marcus

### PR DESCRIPTION
**Pull request by marcusNumminen**
Fixes: #21860

### CODE REVIEW

**Comment by marcusNumminen** [comment]
@microsoft-github-policy-service agree

**Comment by marcusNumminen** [comment]
Ok, the test ui-mode-test-progress.spec.ts runs successfully green when I run it locally....

**Comment by mxschmitt** [comment]
> Ok, the test ui-mode-test-progress.spec.ts runs successfully green when I run it locally....

This one is an unrelated flake, I re-triggered the bots, and now they are green.

**Review by mxschmitt — APPROVED** [review]

**Comment by marcusNumminen** [comment]
Ok. I can't merge the PR due to 'Required status check "license/cla" was not set by the expected GitHub app.' But Under checks it states "All CLA requirements met". See:
image
What should I do?

File: docs/src/trace-viewer.md [hunk 1]
```diff
@@ -168,6 +168,7 @@ await context.tracing.stop({ path: 'trace.zip' });
 
 Available options to record a trace:
 - `'on-first-retry'` - Record a trace only when retrying a test for the first time.
+- `'on-all-retries'` - Record traces only when retrying for all retries.
```
Comment (pavelfeldman) [inline docs/src/trace-viewer.md:171]:
```suggestion
- `'on-all-retries'` - Record traces for all test retries.
```

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**